src/plugins/intel_cpu/src/nodes/mkldnn_roi_align_node.cpp

```diff
@@ -15,14 +15,631 @@
+
+            sub(reg_num_samples, 1);
+
+            jmp(out_loop_label, T_NEAR);
+        }
+        L(out_loop_end_label);
+
+        // store
+        Xbyak::Label store_loop_main_label;
+        Xbyak::Label store_loop_main_end_label;
+        Xbyak::Label store_loop_tail_label;
+        Xbyak::Label store_loop_tail_end_label;
+
+        // EOL for reg_src0-reg_src3
+        reg64_t reg_dst = reg_src0;
+        uni_vpextrq(reg_dst, xmm_args_pool, 1);
+        uni_vpextrq(reg_buf, xmm_args_pool, 0);
+
+        reg64_t reg_dst_stride = reg_src1;
+        if (jcp_.layout == ROIAlignLayoutType::nspc) {
+            int dst_stride = v_step * jcp_.data_size;
+            mov(reg_dst_stride, dst_stride);
+        } else if (jcp_.layout == ROIAlignLayoutType::blk) {
+            int blk_size = (isa == cpu::x64::sse41) ? v_step * 2 : v_step;
+            int dst_stride = blk_size * jcp_.pooled_h * jcp_.pooled_w * jcp_.data_size;
+            mov(reg_dst_stride, dst_stride);
+        }
+
+        L(store_loop_main_label);
+        {
+            cmp(reg_work_amount, v_step);
+            jl(store_loop_main_end_label, T_NEAR);
+
+            load_buf(reg_buf, vmm_buf, v_step);
+            if (jcp_.alg == Algorithm::ROIAlignAvg) {
+                uni_vmulps(vmm_buf, vmm_buf, vmm_scale);
+            }
+            store(vmm_buf, reg_dst, v_step);
+
+            if ((isa == cpu::x64::sse41) && (jcp_.layout == ROIAlignLayoutType::blk)) {
+                add(reg_buf, x_step * sizeof(float));
+                add(reg_dst, x_step * jcp_.data_size);
+
+                load_buf(reg_buf, vmm_buf, x_step);
+                if (jcp_.alg == Algorithm::ROIAlignAvg) {
+                    uni_vmulps(vmm_buf, vmm_buf, vmm_scale);
+                }
+                store(vmm_buf, reg_dst, x_step);
+
+                sub(reg_dst, x_step * jcp_.data_size);
+
+                sub(reg_work_amount, x_step);
+            }
+
+            add(reg_buf, v_step * sizeof(float));
+            add(reg_dst, reg_dst_stride);
+
+            sub(reg_work_amount, v_step);
+
+            jmp(store_loop_main_label, T_NEAR);
+        }
+        L(store_loop_main_end_label);
+
+        int tail_step = 1;
+        L(store_loop_tail_label);
+        {
+            cmp(reg_work_amount, tail_step);
+            jl(store_loop_tail_end_label, T_NEAR);
+
+            load_buf(reg_buf, vmm_buf, tail_step);
+            if (jcp_.alg == Algorithm::ROIAlignAvg) {
+                uni_vmulps(vmm_buf, vmm_buf, vmm_scale);
+            }
+            store(vmm_buf, reg_dst, tail_step);
+
+            add(reg_buf, tail_step * sizeof(float));
+            add(reg_dst, tail_step * jcp_.data_size);
+
+            sub(reg_work_amount, tail_step);
+
+            jmp(store_loop_tail_label, T_NEAR);
+        }
+        L(store_loop_tail_end_label);
+    }
+
+    void get_src(reg64_t& reg_src) {
+        mov(reg_src, reg_src_address);
+        add(reg_src, reg_src_oc);
```

Comment:
maybe I missed smt but why `reg_src_address` and `reg_src_oc` are separate regs? Looks like it is enough just to move reg_src_address each time on sizeof(size_t*)? In fact the code might look as:
```
mov(reg_src0, ptr[reg_src_address** + 0 * sizeof(size_t*)];
mov(reg_src1, ptr[reg_src_address + 1 * sizeof(size_t*)];
mov(reg_src2, ptr[reg_src_address + 2 * sizeof(size_t*)];
mov(reg_src3, ptr[reg_src_address + 3 * sizeof(size_t*)];
add(reg_src_address, 4 * sizeof(size_t*));
```